src/osd/PrimaryLogPG.cc

```diff
@@ -2286,7 +2286,7 @@ void PrimaryLogPG::do_op(OpRequestRef& op)
       return;
   }
 
-  if (obc.get() && obc->obs.exists && obc->obs.oi.has_manifest()) {
+  if (obc.get() && obc->obs.exists) {
```

Comment:
Update this commit message to indicate that maybe_handle_manifest now handles that check.  "make if statement simple" does not make it easier for future readers of this history to understand why this change is correct.  Even better -- squash it into "osd: move check condition into maybe_handle_manifest_detail()".